frontend/app_flowy/packages/appflowy_editor/lib/src/service/internal_key_event_handlers/arrow_keys_handler.dart

```diff
@@ -322,11 +388,30 @@ extension on Position {
       }
       return null;
     }
-    if (node is TextNode) {
-      return Position(path: path, offset: node.delta.nextRunePosition(offset));
-    } else {
-      return Position(path: path, offset: offset);
+    switch (selectionRange) {
+      case _SelectionRange.character:
+        if (node is TextNode) {
```

Comment:
Would it be better if we create an extension for this? Then we can use node.isTextNode()

Reply:
I don't think so. Using `node.isTextNode()` needs to import the extension file. I think that using TextNode will be more concise.

Reply:
Just declare the extension in node.dart file.

Reply:
Good suggestion. However, if we use `node.isTextNode`, then the compiler won't infer the below node is `TextNode`. So we have to use `is TextNode`.

For example.
```dart
class Foo {
  void bar() {}
}

if (object is Foo) {
  object.bar(); // good.
}

if (object.isFoo) {
  object.bar(); // bad.
}
```

Reply:
Cool